feat: Improve auto routing logic in main layout

berk245 opened the pull request:
## Description

This small PR improves the rerouting logic by preventing authenticated users to visit non authenticated routes, such as login and signup. In such scenarios, users are automatically directed to dashboard route. 

## Task

[Notion Task Link](https://www.notion.so/d077118807034bbfb517f36c20d2a304?v=345833d9689145a2b5a197646677c93e&p=080bcc3c55c242f58c32672ef8a560e8&pm=s)

## Screenshots (optional)

berk245 committed:
Improve auto routing logic in main layout

Our existing rerouting flow did not prevent authenticated users
to visit non authenticated routes. With the logic updates,
authenticated users are directed to dashboard automatically.

tonyjojo commented on src/routes/+layout.svelte:
```diff
@@ -28,16 +28,20 @@
+				// Authenticated user wants to go to a public route
 				goto(currentPath);
 				return;
+			} else {
+				return;
```
This block can be removed altogether, right?

berk245 replied:
Yeah, you're right. I wanted to add a default return. Maybe it's not necessary, or at least doesn't have to be in an else block. Removed it for now

tonyjojo commented on src/routes/+layout.svelte:
```diff
@@ -28,16 +28,20 @@
+				goto('/dashboard');
+				return;
+			} else if (user && !noAuthRoutes.includes(currentPath)) {
+				// Authenticated user wants to go to a public route
```
```suggestion
				// Authenticated user wants to go to a private route
```

berk245 replied:
Good catch!

tonyjojo reviewed (commented):
Just being picky here, but could we do `const isPublicRoute = noAuthRoutes.includes(currentPath);` and use it like that?
I find it more elegant and clean

berk245 committed:
Clean up auto rerouting process in main layout

This commit cleans up the process by removing the unnecessarry
if block and introducing a isPublicRoute variable for better
readability.

tonyjojo reviewed (approved):
Good stuff
